src/domain/Project.ts

```diff
@@ -39,14 +39,14 @@ export const getProjectBasePath: (project: Project) => ProjectBasePath = project
   local: ({ basePath }) => basePath,
 });
 
-export const getProjectDir = (project: Project): taskEither.TaskEither<SyncException, ProjectDir> =>
+export const getProjectDir = (project: Project): task.Task<ProjectDir> =>
   pipe(
-    taskEither.Do,
-    taskEither.bind('rootDir', () =>
+    task.Do,
+    task.let('base', () => getProjectBasePath(project)),
+    task.bind('rootDir', () =>
       pipe(
-        api.settingRead('projectDir', NativePathFromStringC),
-        taskEither.fromTaskOption(() => syncExceptionADT.wide.projectDirMissing()),
+        api.settingRead('projectDir', NativePathFromStringC), // FIXME: this is called 'rootDir' now
```

Comment:
It would be great to resolve this FIXME and name it 'rootDir'. Maybe in a follow-up PR?

Reply:
Done!